### PR TITLE
[Nvidia] [DualTor] Skip dscp 2 and 6 in test case test_tunnel_decap_dscp_to_queue_mapping

### PR DESCRIPTION
**Pull request by congh-nvidia**
### Description of PR

Summary:
For Nvidia platforms, the inner dscp 2 and 6 are considered invalid use cases, skip the test. 
The dscp 2 and 6 are reserved only for the outer dscp after the remapping and encapsulation.
We have another similar issue fixed by PR #9625.

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205
- [x] 202305

#### What is the motivation for this PR?
Fix the test failure of test_tunnel_decap_dscp_to_queue_mapping on Nvidia platforms.
#### How did you do it?
 Skip the test for dscp 2 and 6
#### How did you verify/test it?
Run the test case on 4600c dualtor testbed, passed

### CODE REVIEW

**Review by nhe-NV — APPROVED** [review]

**Review by roy-sror — APPROVED** [review]

**Comment by bingwang-ms** [comment]
In current design, DSCP 2/6 is not used at the inner layer in dualtor setup. So the skip makes sense. But can you clarify what's the outer DSCP value for inner DSCP 2/6? And which queue is the traffic mapped to at the active side?

**Comment by bingwang-ms** [comment]
Hi @congh-nvidia . Thanks for the clarification. Please correct me if I missed anything

- Standby side
   Packet with Inner DSCP 2/6 will be  encapsulated with outer DSCP 0 following DSCP_TO_TC_MAP (DSCP 2/6 -> TC 1) and TC_TO_DSCP_MAP (TC 1 -> DSCP 0)

- Active side
Packet with Inner DSCP 2/6 and outer DSCP 0 are being decapsulated and then mapped to TC 1 (following DSCP_TO_TC_MAP|AZRE_TUNNEL) and then mapped to Queue 0 (following TC_TO_QUEUE_MAP).

So the expectation should be the traffic with Inner DSCP 2/6 being mapped to Queue 0 at the active side. Is there something wrong with the expectation?

**Comment by congh-nvidia** [comment]
Hi @bingwang-ms , please see below:

- Standby side

Packet with Inner DSCP 2/6 will be encapsulated with outer DSCP 0 following DSCP_TO_TC_MAP (DSCP 2/6 -> TC 1) and TC_TO_DSCP_MAP (TC 1 -> DSCP 0)

[CH] On Nvidia platform, firstly this is considered as invalid use case. And if the switch receives packets with inner dscp 2/6, they will be mapped to outer dscp 2 and 6 during the encap on standby tor.
This is because the mapping configurations are different on Nvidia platforms when dscp remapping is enabled: https://github.com/sonic-net/sonic-mgmt/blob/master/tests/qos/files/tunnel_qos_map_nvidia.json, and we use the 
 "AZURE_UPLINK" in "DSCP_TO_TC_MAP" for this mapping.


- Active side

Packet with Inner DSCP 2/6 and outer DSCP 0 are being decapsulated and then mapped to TC 1 (following DSCP_TO_TC_MAP|AZRE_TUNNEL) and then mapped to Queue 0 (following TC_TO_QUEUE_MAP).

[CH] In the test, the packet is crafted as 0/2 and 0/6, and it is expected to map it to TC after the decapsulation based on the inner dscp. However, on Nvidia platforms, the expected outer dscp for inner dscp 2/6 are also 2/6, and we do the dscp to TC mapping based on the outer dscp instead of inner dscp. Which means, in the test, because the inner dscps are 2 and 6, according to the logic for Nvidia platforms, it will use TC 2 and 6 to map the queues, not TC1.

Basically, the difference of the mapping configuration and the fact that we do the decap mapping based on outer dscp instead of inner dscp cause the unexpected results in inner dscp 2 and 6. For other dscps, we have already added the nvidia mapping file and specific login in https://github.com/sonic-net/sonic-mgmt/blob/master/tests/qos/tunnel_qos_remap_base.py#L137-L196 to make sure we get the correct expectations.
For inner dscp 2 and 6, since they are invalid use cases, we suggest skip them in the current tests. But if you feel still it's necessary to cover them, we can further adjust the logic for Nvidia platform.

@stephenxs please point out if there is any mistakes in my explanation, thanks.

**Comment by bingwang-ms** [comment]
@congh-nvidia Thanks! I see the gap here. The `DSCP_TO_TC_MAP` is different on Mellanox.

On Mellanox 
```
 "TC_TO_DSCP_MAP": {
        "AZURE_TUNNEL": {
            "0": "8",
            "1": "0",
            "2": "2",
            "3": "2",
            "4": "6",
            "5": "46",
            "6": "6",
            "7": "48",
            "8": "33"
        }
    },
```

Other vendor
```
"TC_TO_DSCP_MAP": {
        "AZURE_TUNNEL": {
            "0": "8",
            "1": "0",
            "2": "0",
            "3": "2",
            "4": "6",
            "5": "46",
            "6": "0",
            "7": "48",
            "8": "33"
        }
    },
```

@stephenxs Do you still remember why we mapped TC 2 to DSCP 2, and TC 6 to DSCP 6?

**Comment by stephenxs** [comment]
> Hi @stephenxs, could you help answer Bing's question? And @bingwang-ms , we have another PR which is opened basically for the same reason as this one: #10419

Yes.
@bingwang-ms 
This is to enable DSCP rewrite logic for the encapsulation packet.

**Review by bingwang-ms — APPROVED** [review]
